Source/Core/JulianDate.js

```diff
@@ -817,15 +827,21 @@ JulianDate.toIso8601 = function (julianDate, precision) {
     .toFixed(precision)
     .replace(".", "")
     .slice(0, precision);
-  return sprintf(
-    "%04d-%02d-%02dT%02d:%02d:%02d.%sZ",
-    year,
-    month,
-    day,
-    hour,
-    minute,
-    second,
-    millisecondStr
+  return (
+    year.toString().padStart(4, "0") +
+    "-" +
+    month.toString().padStart(2, "0") +
+    "-" +
+    day.toString().padStart(2, "0") +
+    "T" +
+    hour.toString().padStart(2, "0") +
+    ":" +
+    minute.toString().padStart(2, "0") +
+    ":" +
+    second.toString().padStart(2, "0") +
+    "." +
+    millisecondStr +
+    "Z"
```

Comment:
Not sure if it'd make more sense to wrap these in a separate function - I tried a small wrapper around `padStart` to avoid all the calls to `toString()` but since there were only two use cases and both have different syntax it didn't look much cleaner.

Reply:
I agree with your choice here, we don't really gain anything by string to add a helper function here.